tests/foreman/cli/test_ansible.py

```diff
@@ -732,3 +734,291 @@ def _finalize():
         )
         result = target_sat.cli.JobInvocation.info({'id': job_id})['success']
         assert result == '1'
+
+
+@pytest.mark.upgrade
+class TestAnsibleAAPIntegration:
+    """Test class for Satellite integration with Ansible Automation Controller
+
+    :CaseComponent: Ansible-ConfigurationManagement
+    """
+
+    def update_sat_credentials_in_aap(
+        self,
+        aap_client,
+        sat,
+        username=settings.server.admin_username,
+        password=settings.server.admin_password,
+        creds_name=settings.AAP_INTEGRATION.satellite_credentials,
+    ):
+        # Find the Satellite credentials in AAP and update it for target_sat.hostname and user credentials
+        creds_list = aap_client.get(
+            '/api/v2/credentials/', query_parameters=f'name={creds_name}'
+        ).json()
+        new_creds = {
+            'inputs': {
+                'host': f'https://{sat.hostname}',
+                'username': username,
+                'password': password,
+            }
+        }
+        response = aap_client.patch(
+            f'/api/v2/credentials/{creds_list["results"][0]["id"]}/', json=new_creds
+        )
+        assert response.ok
+
+    @pytest.fixture(scope='class')
+    def aap_client(self):
+        client = awxkit.api.client.Connection(f'https://{settings.AAP_INTEGRATION.AAP_FQDN}/')
+        client.login(settings.AAP_INTEGRATION.USERNAME, settings.AAP_INTEGRATION.PASSWORD)
+        yield client
+        client.logout()
+
+    @pytest.mark.parametrize('auth_type', ['admin', 'non-admin'])
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    def test_positive_ansible_dynamic_inventory(
+        self,
+        request,
+        target_sat,
+        module_org,
+        module_location,
+        module_ak_with_cv,
+        aap_client,
+        rhel_contenthost,
+        auth_type,
+    ):
+        """Verify AAP is able to pull a dynamic inventory of hosts from Satellite,
+        with admin and non-admin user.
+
+        :id: ad25aee5-4ea3-4743-a301-1c6271856g19
+
+        :steps:
+            1. Register content host to Satellite with admin and non-admin user
+            2. Update Satellite type credentials in AAP
+            3. Find Satellite inventory, inventory source and sync inventory source in AAP
+            4. Verify registered hosts are added to Satellite inventory
+
+        :expectedresults: All hosts managed by Satellite are added to Satellite inventory.
+        """
+        inventory_name = settings.AAP_INTEGRATION.satellite_inventory
+
+        password = settings.server.admin_password
+        if auth_type == 'admin':
+            login = gen_string('alpha')
+            user = target_sat.api.User(
```

Comment:
this could as well be unified.

We can construct the `params` dict, we update it by `admin` and `roles` keys respectively and then call `user = target_sat.api.User(**params)`

Reply:
I feel, it would make code less readable for now, but I'll consider it when I move this part into a new user setup fixture in a separate PR as discussed earlier